contrib/babelfishpg_tds/src/backend/tds/tdsutils.c

```diff
@@ -925,6 +927,35 @@ check_babelfish_droprole_restrictions(char *role)
 	}
 }
 
+/*
+ * Checks if bbf_role_admin is the direct admin of a role.
+ */
+static bool
+is_bbf_admin_of_role(Oid roleid, Oid bbf_role_admin)
```

Comment:
I think we should make this function generic by naming it like `is_member_immediate_admin_of_role` and then checking whether given member is immediate admin of role or not.